### DISCUSSION TITLE
Easy delete and copy/paste frames (+ other ideas)

### DISCUSSION THREAD

**Discussion by BaoGM** [ggarra13/mrv2#82] [Ideas]
Hey there !

I'm a cg animator and I was looking for a free open source professional player when I stumble upon yours.

It does a really great job but I have nonetheless some requests which would greatly improve the use of your player for animator use. I hope you will find this useful :

- Add a delete frame (with sound) option/hotkey.
Quicktime professional player got that built in option and it's so far the only player I've found having that very, very useful possibility even rv player doesn't have. For a professional animator it is a real game changer. If I would have only one request it would be this one.

- Add a copy/paste frame option/hotkey.
Basically a copy/paste frame. Again, for animator use it's great.

- Draw on multiple frames / extend current drawing to multiple frames
Currently it's possible to draw on the current frames, on all frames or to extend ghosting to several frames before/after. But there is no options to draw on a set number of frames or to draw on one frame then extend it to other frames.

- Moving drawing tool
Would be great if we could move a drawing or text somewhere else once drawn.

Thank you for your time and tank you for your great work !

**Comment by ggarra13** [comment]
While mrv2 does not have an Edit feature yet, you will be able to see the OpenTimelineIO timeline before the end of the year.   Editing is something I wish to have, but there's no ETA for it yet.  You might want to take a look at xStudio later this year, as they are also promising an editing functionality.  Or my old mrViewer in sourceforge, which has a rather horrible but functional EDL Edit functionality.

If you want to delete a frame or copy a frame (or set of frames) **only once** right now, you **can** currently do it in mrv2 with the Playlist panel.  You do the following:

- In the Files Panel, you select the clip you want to delete a frame.
- You mark the in and out points to where you want the clip to start (and where you want the cut to end -- ie. before the frame you want to remove).   You can use the I and O shortcuts to mark in and out points.
- You go to the Playlist Panel and click on "Add".
- You select the clip again in the FIles Panel and using the RMB on the clip, select File->Clone.
- You select the new cloned clip and mark the in point after the frame you want to delete and the out where you want it to end.
- You click on "Add" again on the Playlist Panel.
- You click on "Create" to create a temporary .otio timeline on a temp directory.
- Play back the new .otio file to see your cut.

As I said, you can only do this once, as the resulting .otio file cannot currently be further edited.

BTW, I am curious.  While I understand the reason for editing in the cut, how do you use the removing of a frame in a concrete case?

**Comment by BaoGM** [comment]
Concretely, you'll retime your animation by first deleting on the go expendables inbetween in the player instead of doing it in the animation software.

For instance, you do your draft animation, do a quick playblast and easily retime it on the player by deleting or holding frames. Once you're happy with the result, you can return to your animation software and report back what you did in that player.

It's by far a smoother process doing it that way instead of doing it the hard way, by trial and errors directly in a laggy animation software where you need to move numerous keys on numerous objects at the same time. :D

The keywords here are 'on the go' ; so far sadly there is to my knowledge only the outdated quicktime player (pro version) which have that amazing for animation yet nearly unknown feature.

Thanks for your quick answer anyway !

**Comment by ggarra13** [comment]
mrv2 v0.7.5 adds the Edit/Frame/Cut, Edit/Frame/Copy, Edit/Frame/Paste and Edit/Frame/Insert functionality.  Be sure to try it out and report back how it works for you, bugs you find, etc.

**Comment by BaoGM** [comment]
Hey thank you for the update! I'm really glad you heard my recommendations,
I'm gonna try it tomorrow!

_

Gabriel Montaner

https://www.linkedin.com/in/gabriel-montaner-0b310b18b

Le sam. 9 sept. 2023, 19:32, Gonzalo Garramuño ***@***.***> a
écrit :

**Comment by ggarra13** [comment]
Please make sure you try v0.7.6 as I found out some memory corruption issues on Linux and macOS.
